--- node_resource.inc.orig	2014-09-24 14:24:44.000000000 -0400
+++ node_resource.inc	2014-11-06 02:08:36.640525709 -0500
@@ -38,6 +38,37 @@
           'access arguments' => array('create'),
           'access arguments append' => TRUE,
         ),
+        'translate' => array(
+          'help' => 'Translate a node',
+          'file' => array('type' => 'inc', 'module' => 'services', 'name' => 'resources/node_resource'),
+          'callback' => '_node_resource_translate',
+          'args' => array(
+            array(
+              'name' => 'nid',
+              'optional' => FALSE,
+              'source' => array('path' => 0),
+              'type' => 'int',
+              'description' => 'The nid of the node to translate',
+            ),
+            array(
+              'name' => 'values',
+              'optional' => FALSE,
+              'source' => array('data' => 'values'),
+              'type' => 'array',
+              'description' => 'The node data to add to translation',
+            ),
+            array(
+              'name' => 'language',
+              'optional' => FALSE,
+              'source' => array('data' => 'language'),
+              'type' => 'string',
+              'description' => 'The language to translate to',
+            ),
+          ),
+          'access callback' => '_node_resource_access',
+          'access arguments' => array('update'),
+          'access arguments append' => TRUE,
+        ),
         'update' => array(
           'help' => 'Update a node',
           'file' => array('type' => 'inc', 'module' => 'services', 'name' => 'resources/node_resource'),
@@ -318,7 +349,10 @@
   $stub_form = drupal_get_form($node_type . '_node_form', (object) $stub_node);
   $form_state['triggering_element'] = $stub_form['actions']['submit'];
 
-  drupal_form_submit($node_type . '_node_form', $form_state, (object)$stub_node);
+  // Change $stub_node to $node in call to drupal_form_submit()
+  // below in order to handle multi-value fields
+  //drupal_form_submit($node_type . '_node_form', $form_state, (object)$stub_node);
+  drupal_form_submit($node_type . '_node_form', $form_state, (object)$node);
 
   if ($errors = form_get_errors()) {
     return services_error(implode(" ", $errors), 406, array('form_errors' => $errors));
@@ -356,6 +390,26 @@
   }
 }
 
+function _node_resource_translate($nid, $values, $new_lang) {
+  $node = node_load($nid);
+
+  $handler = entity_translation_get_handler('node', $node, TRUE);
+
+  $translation = array(
+    'translate' => 0,
+    'status' => 1,
+    'language' => $new_lang, // here is the language you're translating to
+    'source' => 'en',        // here is the source language
+  );
+
+  $handler->setTranslation($translation, $values);
+
+  field_attach_update('node', $node);
+#   node_save($node);
+
+  return TRUE;
+}
+
 /**
  * Updates a new node based on submitted values.
  *
@@ -402,6 +456,10 @@
   $node_type = $node['type'];
   node_object_prepare($old_node);
 
+  if ($node['tnid']) {
+    $old_node->tnid = $node['tnid'];
+  }
+
   // Setup form_state.
   $form_state = array();
   $form_state['values'] = $node;
